pkg/disruption/backend/disruption/handler.go

```diff
@@ -73,11 +74,11 @@ func (h *ciHandler) Unavailable(from, to *backend.SampleResult) {
 	klog.V(4).Info(message)
 	h.eventRecorder.Eventf(
 		&v1.ObjectReference{Kind: "OpenShiftTest", Namespace: "kube-system", Name: h.descriptor.Name()},
-		nil, v1.EventTypeWarning, string(eventReason), "detected", message)
+		nil, v1.EventTypeWarning, string(eventReason), "detected", message.BuildString())
 
-	condition := monitorapi.NewInterval(monitorapi.SourceDisruption, level).Locator(h.descriptor.DisruptionLocator()).
-		Message(monitorapi.NewMessage().HumanMessage(message).Reason(eventReason)).BuildCondition()
-	openIntervalID := h.monitorRecorder.StartInterval(fs.StartedAt, condition)
+	interval := monitorapi.NewInterval(monitorapi.SourceDisruption, level).Locator(h.descriptor.DisruptionLocator()).
+		Message(message).Build(fs.StartedAt, time.Time{})
+	openIntervalID := h.monitorRecorder.StartInterval(interval)
```

Comment:
Disruption intervals are the most complicated part in here, but I think I found it was workable by keeping the odd recorder.StartInterval / recorder.EndInterval(index) alive, which is lower risk than trying to port it all to straight AddInterval.